scripts/manage_demux_stats.py

```diff
@@ -89,76 +90,122 @@ def manipulate_process(demux_process, proc_stats):
+                        if lane_no in undet_lanes:
+                            undet_included = True
+                            #Sanity check for including undetermined
+                            #Next entry is undetermined and previous is for a different lane
+                            current_index = parser_struct.index(entry)
+                            undet = parser_struct[current_index + 1]
+                            if undet['Sample'] == 'Undetermined' and parser_struct[current_index - 1]['Lane'] != lane_no:
+                                try:
+                                    clusterType = None
+                                    if "PF Clusters" in undet:
+                                        clusterType = "PF Clusters"
+                                    else:
+                                        clusterType = "Clusters"
+                                    #Paired runs are divided by two within flowcell parser
+                                    if proc_stats["Paired"]:
+                                        undet_reads = int(undet[clusterType].replace(",",""))*2
+                                        undet_pairs = int(undet[clusterType].replace(",",""))
+                                    #Since a single ended run has no pairs, pairs is set to equal reads
+                                    else:
+                                        undet_reads = int(undet[clusterType].replace(",",""))
+                                        undet_pairs = int(undet[clusterType].replace(",",""))
+                                    logger.info("Included undetermined for lane number {}".format(lane_no))
+                                except Exception as e:
+                                    problem_handler("exit", "Unable to set values for undetermined #Reads and #Read Pairs: {}".format(e.message))
+                        
                     if sample == current_name:
                         logger.info("Added the following set of values to {} of lane {}:".format(sample,lane_no))
                         try:
                             target_file.udf["%PF"] = float(entry["% PFClusters"])
                             logger.info("{}% PF".format(str(target_file.udf["%PF"])))
-                            target_file.udf["% One Mismatch Reads (Index)"] = float(entry["% One mismatchbarcode"])
-                            logger.info("{}% One Mismatch Reads (Index)".format(str(target_file.udf["% One Mismatch Reads (Index)"])))
+                            
+                            #["% One mismatchbarcode"] can hold NaN. Treating it as 0.0
+                            if entry["% One mismatchbarcode"] == "NaN":
+                                target_file.udf["% One Mismatch Reads (Index)"] = 0.0
+                                logger.info("'NaN' One Mismatch Reads (Index), treating as {}".format(str(target_file.udf["% One Mismatch Reads (Index)"])))
```

Comment:
the plus side of using format is that you don't need to cast stuff to string to include them in a string.